clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3495,6 +3493,31 @@ private ReadyToServeCheck getDefaultReadyToServeChecker() {
     };
   }
 
+  void reportCompletedAndSendHeartBeat(PartitionConsumptionState partitionConsumptionState, int partition) {
+    reportCompletedAndSendHeartBeat(partitionConsumptionState, partition, false);
+  }
+
+  /**
+   * Whenever a leader is marked to be completed, it should send a heartbeat SOS to VT
+   * check {@link StoreIngestionTask#sendInstantHeartBeatAndUpdateVeniceWriter} for more details.
+   */
+  void reportCompletedAndSendHeartBeat(
+      PartitionConsumptionState partitionConsumptionState,
+      int partition,
+      boolean forceCompletion) {
+    statusReportAdapter.reportCompleted(partitionConsumptionState, forceCompletion);
+    LOGGER.info("{} Partition {} is ready to serve", consumerTaskId, partition);
+    if (partitionConsumptionState.getLeaderFollowerState().equals(LeaderFollowerStateType.LEADER)) {
+      sendInstantHeartBeatAndUpdateVeniceWriter(
+          partitionConsumptionState,
+          new PubSubTopicPartitionImpl(versionTopic, partition));
+    }
+  }
+
+  protected abstract void sendInstantHeartBeatAndUpdateVeniceWriter(
```

Comment:
I feel the structure is not quite right..

Even this function is called `reportCompletedAndSendHeartBeat`, it is not guaranteed that the HB will be sent out since there is a `leader` state check.

Shall we just call it `reportCompleted` and let `LFSIT` override this function and internally it will decide whether it should send out a HB or not.
With this way, we don't need this function any more: `sendInstantHeartBeatAndUpdateVeniceWriter`.

Reply:
Great suggestion. cleaned it up.